SUPPORTED_APIS.md

```diff
@@ -2,31 +2,122 @@
+| `HARDHAT` | `hardhat_impersonateAccount` | NOT IMPLEMENTED | Impersonate an account |
+| `HARDHAT` | `hardhat_getAutomine` | NOT IMPLEMENTED | Returns true if automatic mining is enabled, and false otherwise |
+| `HARDHAT` | `hardhat_metadata` | NOT IMPLEMENTED | Returns the metadata of the current hardhat network |
+| `HARDHAT` | `hardhat_mine` | NOT IMPLEMENTED | Can mine any number of blocks at once, in constant time |
+| `HARDHAT` | `hardhat_reset` | NOT IMPLEMENTED | Resets the state of the network |
+| `HARDHAT` | `hardhat_setBalance` | NOT IMPLEMENTED | Enables or disables automatic mining |
+| `HARDHAT` | `hardhat_setCode` | NOT IMPLEMENTED | Sets the bytecode of a given account |
+| `HARDHAT` | `hardhat_setCoinbase` | NOT IMPLEMENTED | Sets the coinbase address |
+| `HARDHAT` | `hardhat_setLoggingEnabled` | NOT IMPLEMENTED | Enables or disables logging |
+| `HARDHAT` | `hardhat_setMinGasPrice` | NOT IMPLEMENTED | Sets the minimum gas price |
+| `HARDHAT` | `hardhat_setNextBlockBaseFeePerGas` | NOT IMPLEMENTED | Sets the base fee per gas for the next block |
+| `HARDHAT` | `hardhat_setPrevRandao` | NOT IMPLEMENTED | Sets the RANDAO value of the next block |
+| `HARDHAT` | `hardhat_setNonce` | NOT IMPLEMENTED | Sets the nonce of a given account |
+| `HARDHAT` | `hardhat_setStorageAt` | NOT IMPLEMENTED | Sets the storage value at a given key for a given account |
+| `HARDHAT` | `hardhat_stopImpersonatingAccount` | Stop impersonating an account after having previously used |
+| `EVM` | `evm_addAccount` | NOT IMPLEMENTED | Adds any arbitrary to the namespace |
+| `EVM` | `evm_increaseTime` | NOT IMPLEMENTED | Jump forward in time by the given amount of time |
+| `EVM` | `evm_mine` | NOT IMPLEMENTED | Force a single block to be mined |
+| `EVM` | `evm_removeAccount` | NOT IMPLEMENTED | Removes an account from the namespace |
+| `EVM` | `evm_revert` | NOT IMPLEMENTED | Revert the state of the blockchain to a previous snapshot |
+| `EVM` | `evm_setAccountBalance` | NOT IMPLEMENTED | Sets the given account's balance to the specified WEI value |
+| `EVM` | `evm_setAccountCode` | NOT IMPLEMENTED | Sets the given account's code to the specified data |
+| `EVM` | `evm_setAccountNonce` | NOT IMPLEMENTED | Sets the given account's nonce to the specified value |
+| `EVM` | `evm_setAccountStorageAt` | NOT IMPLEMENTED | Sets the given account's storage slot to the specified data |
+| `EVM` | `evm_setTime` | NOT IMPLEMENTED | Sets the internal clock time to the given timestamp |
+| `EVM` | `evm_snapshot` | NOT IMPLEMENTED | Snapshot the state of the blockchain at the current block |
+| `ETH` | `eth_accounts` | NOT IMPLEMENTED | Returns a list of addresses owned by client |
+| `ETH` | `eth_blockNumber` | NOT IMPLEMENTED | Returns the number of the most recent block |
+| `ETH` | `eth_call` | NOT IMPLEMENTED | Executes a new message call immediately without creating a transaction on the block chain |
```

Comment:
This one is implemented: https://github.com/matter-labs/era-test-node/blob/main/src/node.rs#L474

Reply:
This ones also confusing. Running:

```bash
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "2",
    "method": "eth_call",
    "params": [{
        "to": "0x0000",
        "data": "0x0000",
        "from": "0x0000",
        "gas": "0x0000",
        "gasPrice": "0x0000",
        "value": "0x0000",
        "nonce": "0x0000"
    }, "latest"]
}
```

returns:

```json
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params: invalid length 4, expected a (both 0x-prefixed or not) hex string with length of 40."
  },
  "id": "2"
}
```

And:

```bash
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "2",
    "method": "eth_call",
    "params": ["0x0000", "latest"]
}
```

returns:

```json
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params: invalid type: string \"0x0000\", expected struct CallRequest."
  },
  "id": "2"
}
```

Reply:
Yeah, for `eth_call`, you can validate this with any `CALL` to the network. For example, if you deploy the `Greeter.sol` smart contract, calling `await contract.greet()` will make a request to `eth_call`